src/js/pages/network/virtual-network-detail/VirtualNetworkTaskTab.js

```diff
@@ -0,0 +1,291 @@
+      <colgroup>
+        <col style={{width: '50%'}} />
+        <col />
+        <col />
+      </colgroup>
+    );
+  }
+
+  getEmptyScreen() {
+    return (
+      <AlertPanel
+        title="Virtual Network Not Found"
+        iconClassName="icon icon-sprite icon-sprite-jumbo icon-sprite-jumbo-white icon-network flush-top">
+        <p className="flush">
+          {'Could not find the requested virtual network. Go to '}
+          <Link to="virtual-networks-tab">
+            Virtual Networks
+          </Link> overview to see all virtual networks.
+        </p>
+      </AlertPanel>
+    );
+  }
+
+  getErrorScreen() {
+    return <RequestErrorMsg />;
+  }
+
+  getLoadingScreen() {
+    return (
+      <div className="container container-fluid container-pod text-align-center vertical-center inverse">
+        <div className="row">
+          <div className="ball-scale">
+            <div />
+          </div>
+        </div>
+      </div>
+    );
+  }
+
+  getTaskLink(taskID, value, title) {
+    if (!value) {
+      value = taskID;
+    }
+
+    if (!title) {
+      title = taskID;
+    }
+
+    return (
+      <Link
+        className="clickable"
+        params={{overlayName: this.props.overlay.getName(), taskID}}
+        title={title}
+        to="virtual-networks-tab-detail-tasks-detail">
+        {value}
+      </Link>
+    );
+  }
+
+  getTitle(portMappings) {
+    return portMappings.map(function (mapping) {
+      return `${mapping.container_port} > ${mapping.host_port} (${mapping.protocol})`
+    }).join(', ')
+  }
+
+  renderHeading(prop) {
+    return (
+      <span className="table-header-title">{headerMapping[prop]}</span>
+    );
+  }
+
+  renderID(prop, task) {
+    return this.getTaskLink(task.id);
+  }
+
+  renderPorts(prop, task) {
+    let portMappings = TaskUtil.getPortMappings(task);
+    if (!portMappings) {
+      return 'N/A';
+    }
+
+    let title = this.getTitle(portMappings);
+    if (portMappings.length > 3) {
+      portMappings = portMappings.slice(0, 2);
+      portMappings.push({container_port: '...'});
+    }
+
+    let {id} = task;
+    return portMappings.map((mapping, index) => {
+      let mapTo = [];
+
+      if (mapping.host_port) {
+        mapTo.push(
+          <i className="icon icon-sprite icon-sprite-mini icon-chevron flush" />
+        );
+        mapTo.push(this.getTaskLink(
+          id,
+          `${mapping.host_port} (${mapping.protocol})`,
+          title
+        ));
+      }
+
+      return (
+        <div key={index} className="table-cell-value">
+          <div className="table-cell-details-secondary flex-box flex-box-align-vertical-center table-cell-flex-box">
+            <div className="text-overflow service-link inverse">
+              {this.getTaskLink(id, mapping.container_port, title)}
+              {mapTo}
+            </div>
+          </div>
+        </div>
+      );
+    });
+  }
+
+  render() {
+    let {searchString} = this.state;
+    if (this.isLoading()) {
```

Comment:
I don't think you really need the `#isLoading` function, as you only use it once.